src/Avalonia.Themes.Fluent/Accents/SystemAccentColors.cs

```diff
@@ -0,0 +1,163 @@
+            {
+                EnsureColors();
+                value = _systemAccentColorDark1;
+                return true;
+            }
+
+            if (strKey.Equals(AccentDark2Key, StringComparison.InvariantCulture))
+            {
+                EnsureColors();
+                value = _systemAccentColorDark2;
+                return true;
+            }
+
+            if (strKey.Equals(AccentDark3Key, StringComparison.InvariantCulture))
+            {
+                EnsureColors();
+                value = _systemAccentColorDark3;
+                return true;
+            }
+
+            if (strKey.Equals(AccentLight1Key, StringComparison.InvariantCulture))
+            {
+                EnsureColors();
+                value = _systemAccentColorLight1;
+                return true;
+            }
+
+            if (strKey.Equals(AccentLight2Key, StringComparison.InvariantCulture))
+            {
+                EnsureColors();
+                value = _systemAccentColorLight2;
+                return true;
+            }
+
+            if (strKey.Equals(AccentLight3Key, StringComparison.InvariantCulture))
+            {
+                EnsureColors();
+                value = _systemAccentColorLight3;
+                return true;
+            }
+        }
+
+        value = null;
+        return false;
+    }
+
+    public IResourceHost? Owner { get; private set; }
+    public event EventHandler? OwnerChanged;
+    public void AddOwner(IResourceHost owner)
+    {
+        if (Owner != owner)
+        {
+            Owner = owner;
+            OwnerChanged?.Invoke(this, EventArgs.Empty);
+
+            if (_platformSettings is not null)
+            {
+                _platformSettings.ColorValuesChanged += PlatformSettingsOnColorValuesChanged;
+            }
+        }
+    }
+
+    public void RemoveOwner(IResourceHost owner)
+    {
+        if (Owner == owner)
+        {
+            Owner = null;
+            OwnerChanged?.Invoke(this, EventArgs.Empty);
+            
+            if (_platformSettings is not null)
+            {
+                _platformSettings.ColorValuesChanged -= PlatformSettingsOnColorValuesChanged;
+            }
+        }
+    }
+
+    private void EnsureColors()
+    {
+        if (_invalidateColors)
+        {
+            _invalidateColors = false;
+            
+            _systemAccentColor = _platformSettings?.GetColorValues().AccentColor1 ?? s_defaultSystemAccentColor;
+            (_systemAccentColorDark1,_systemAccentColorDark2, _systemAccentColorDark3,
+                    _systemAccentColorLight1, _systemAccentColorLight2, _systemAccentColorLight3) = CalculateAccentShades(_systemAccentColor);
+        }
+    }
+
+    public static (Color d1, Color d2, Color d3, Color l1, Color l2, Color l3) CalculateAccentShades(Color accentColor)
+    {
+        // dark1step = (hslAccent.L - SystemAccentColorDark1.L) * 255
```

Comment:
If anybody interested, I was able to replicate close to identical algorithm of getting accent color shades.
(unless it's different depending on the color)

Reply:
A few years ago some of us tried to figure this out as well. Nobody thought at the time to use HSL but we did use HSV. In that HSV analysis weightings were changed based on the color. The algorithm seems adaptive to perception for each color. Maybe HSL really is the solution though.

Here is some of the old discussion.
https://github.com/CommunityToolkit/WindowsCommunityToolkit/issues/3363#issuecomment-650562951

Whatever is done here for the Fluent color palette really should also be done in the ColorPicker's previewer control. The intention was for that to work the same as what you are doing here.